### ISSUE TITLE
Global search should search opened documents

### ISSUE THREAD

**Issue by pathwave** [helix-editor/helix#4207]
Currently global search only read files from disk, this creates two problems:
Unsaved content is not search-able and the resulting search result can end up on wrong line (and selection) because of unsaved content in-between. I think it would be better if it searched throught opened documents and excluded those opened documents from the disk search.

**Comment by pppKin** [comment]
Good call. I think this would be easier to implement after (the much anticipated) #3110 is merged.

**Comment by pathwave** [comment]
I implemented it for #4381, so it just needs a copy/paste ish after im done with #4381.
